dhis-2/dhis-services/dhis-service-tracker/src/test/java/org/hisp/dhis/tracker/imports/validation/validator/enrollment/DateValidatorTest.java

```diff
@@ -168,4 +174,39 @@ void testFailOnMissingOccurredAtDate() {
 
     assertHasError(reporter, enrollment, E1023);
   }
+
+  @Test
+  void shouldFailWhenCompletedAtIsPresentAndStatusIsNotCompleted() {
+    Enrollment enrollment = new Enrollment();
+    enrollment.setEnrollment(CodeGenerator.generateUid());
+    enrollment.setProgram(MetadataIdentifier.ofUid(CodeGenerator.generateUid()));
+    enrollment.setOccurredAt(now());
+    enrollment.setCompletedAt(now());
+    enrollment.setStatus(EnrollmentStatus.ACTIVE);
+
+    when(preheat.getProgram(enrollment.getProgram())).thenReturn(new Program());
+
+    validator.validate(reporter, bundle, enrollment);
+
+    assertHasError(reporter, enrollment, E1052);
+  }
+
+  @ParameterizedTest
+  @EnumSource(
+      value = EnrollmentStatus.class,
+      names = {"COMPLETED", "CANCELLED"})
+  void shouldValidateWhenCompletedAtIsPresentAndStatusAcceptCompletedAt(EnrollmentStatus status) {
```

Comment:
```suggestion
  void shouldPassWhenCompletedAtIsPresentAndStatusAcceptCompletedAt(EnrollmentStatus status) {
```

is opposed to the `shouldFail` above (not sure if other validator tests already have a naming scheme/word we can reuse).